src/setup-python.ts

```diff
@@ -61,12 +61,15 @@ function resolveVersionInput(): string {
 }
 
 async function run() {
-  // According to the README windows binaries do not require to be installed
-  // in the specific location, but Mac and Linux do
-  if (!IS_WINDOWS && !process.env.AGENT_TOOLSDIRECTORY?.trim()) {
-    if (IS_LINUX) process.env['AGENT_TOOLSDIRECTORY'] = '/opt/hostedtoolcache';
-    else process.env['AGENT_TOOLSDIRECTORY'] = '/Users/runner/hostedtoolcache';
+  // When setting AGENT_TOOLSDIRECTORY, the actions/tool-cache function find
+  // is not able to find the files cached by actions/python-version.
+  if (process.env.AGENT_TOOLSDIRECTORY?.trim() && !IS_MAC) {
     process.env['RUNNER_TOOL_CACHE'] = process.env['AGENT_TOOLSDIRECTORY'];
+  } else if (process.env.AGENT_TOOLSDIRECTORY?.trim() && IS_MAC) {
+    process.env['AGENT_TOOLSDIRECTORY'] = process.env['RUNNER_TOOL_CACHE'];
+    core.warning(
+      'AGENT_TOOLSDIRECTORY is not valid for MacOS as shared libraries are configured with a fixed path.'
+    );
```

Comment:
I'd better suggest using this logic. If there is no `AGENT_TOOLSDIRECTORY` and it is MacOS - > assign the default `/Users/runner/hostedtoolcache` in order not to have any problems with shared libraries. Second If statement is for proper search of the tool cache location on the runner (now we search it in `RUNNER_TOOL_CACHE`).
```suggestion
  if (!process.env.AGENT_TOOLSDIRECTORY?.trim() && IS_MAC) {
    process.env['AGENT_TOOLSDIRECTORY'] = '/Users/runner/hostedtoolcache';
  }
  
  if (process.env.AGENT_TOOLSDIRECTORY?.trim()){
    process.env['RUNNER_TOOL_CACHE'] = process.env['AGENT_TOOLSDIRECTORY'];
  }

  core.debug(
    `Python is expected to be installed into ${
      process.env.AGENT_TOOLSDIRECTORY?.trim()
        ? process.env['AGENT_TOOLSDIRECTORY']
        : process.env['RUNNER_TOOL_CACHE']
    }`
  );
```

Reply:
Oh yes, that's  much nicer! My TypeScript is pretty rusty these days :)